src/Core/System/Snippet/SnippetService.php

```
@@ -314,12 +323,25 @@ private function getSnippetFilesByIso(array $isoList): array
     {
         $result = [];
         foreach ($isoList as $iso) {
-            $mappedIso = match ($iso) {
-                'de-DE' => 'de',
-                'en-GB' => 'en',
-                default => $iso,
-            };
-            $result[$iso] = $this->snippetFileCollection->getSnippetFilesByIso($mappedIso);
+            // Load all snippet files that match the exact locale (e.g., "es-AR")
+            $files = $this->snippetFileCollection->getSnippetFilesByIso($iso);
+            preg_match(
+                SnippetPatterns::COMPLETE_LOCALE_PATTERN,
+                $iso,
+                $matchedPattern,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            // If the locale has a region (e.g., "es-AR"), try to load its base language ("es")
+            if (!empty($matchedPattern['region']) && strtolower($matchedPattern['region']) !== $iso) {
+                \assert(!empty($matchedPattern['language']));
```

`assert` is more of a dev tool and is not a real error handling in production. You should throw an exception if language is not set.

---

I removed the asserts and added a check in the if. I dont think we need an exception, we can just skip adding the base file if the language is not set.